src/main/java/pong/Ball.java

```diff
@@ -1,4 +1,65 @@
 package pong;
 
 public class Ball {
+    private Paddle userPaddle;
+    private Paddle compPaddle;
+
+    private double x;
+    private double y;
+    private double velX;
+    private double velY;
+    private final int MID_OF_BALL = PongView.BALL_WIDTH / 2;
+
+    public Ball(Paddle userPaddle, Paddle compPaddle) {
+        //first position - middle of the board.
+        x = Environment.WIDTH / 2;
+        y = Environment.HEIGHT / 2;
+
+        //random velocity to start the game off
+        velY = getRandomVelocity();
+        velX = getRandomVelocity();
+        this.userPaddle = userPaddle;
+        this.compPaddle = compPaddle;
+    }
+
+    private double getRandomVelocity() {
+        //randSpeed can be any number 2-4
+        double randSpeed = (Math.random() * 3 + 2);
+        double randDirection = Math.random() * 2;
+        //randDirection will be assigned as either 1 or -1
+        randDirection = randDirection == 1? 1: -1;
+
+        return randSpeed * randDirection;
+    }
+    public int getX() {
+        return (int) x;
+    }
+
+    public int getY() {
+        return (int) y;
+    }
+
+    public void move() {
+        x += velX;
+        y += velY;
+        checkPaddleCollide();
+        checkWallCollide();
+    }
+
+    public void checkWallCollide() {
+        if (this.y >= PongFrame.HEIGHT - MID_OF_BALL || this.y <= MID_OF_BALL)
+            velX = -velX;
```

Comment:
velX and velY are the directions the ball is going in, right? Wouldn't it make sense the if you hit the top or bottom wall, it is the up and down direction that gets reversed. So should this be velY? And in checkPaddleCollide() you should be changing velX?

Reply:
Yes! You are so right! Thank you for catching this

Reply:
Fixed this in 40240ba